components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/AuthenticationRequestBuilder.java

```diff
@@ -159,4 +167,15 @@ private List<AllowedOperation> getAllowedOperations() {
         operation.setOp(Operation.REDIRECT);
         return Collections.singletonList(operation);
     }
+
+    private Request getRequest(HttpServletRequest httpServletRequest) {
```

Comment:
IMO, we should get rid of using servlet request context at this level. Why can't we use IdentityContext now. That's the whole purpose to load request headers and params for IdentityContext.
Anyway we have to verify few things. Given IdentityContext is a thread local at a high request load if it's not properly cleared an all there can be problems. But this is same for carbon context. If it's being utilized in login flow I think we can rely on IdentityContext here